src/redux/books/books.js

```diff
@@ -1,47 +1,82 @@
-import { v4 as uuidv4 } from 'uuid';
+// import { v4 as uuidv4 } from 'uuid';
 
 // Actions
+const GET_BOOKS = 'bookstore-react/books/GET_BOOKS';
 const ADD = 'bookstore-react/books/ADD';
 const REMOVE = 'bookstore-react/books/REMOVE';
 
 // Reducer
-export default function booksReducer(
-  state = [
-    {
-      title: 'First Book',
-      author: 'First Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Second Book',
-      author: 'Second Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Third Book',
-      author: 'Third Author',
-      id: uuidv4(),
-    },
-  ],
-  action = {}
-) {
+export default function booksReducer(state = [], action = {}) {
   switch (action.type) {
+    case GET_BOOKS:
+      return action.payload;
     case ADD:
       return [...state, action.book];
     case REMOVE:
-      return state.filter((book) => book.id !== action.book.id);
+      return state.filter((book) => book.item_id !== action.book.item_id);
     default:
       return state;
   }
 }
 
 // Action Creators
-export const addBook = (book) => ({
-  type: ADD,
-  book,
-});
+export const getBooks = () => async (dispatch) => {
+  const booksFetch = await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/z4gzcApO94uwV3qBSQiu/books',
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    }
+  ).then((data) => data.json());
 
-export const removeBook = (book) => ({
-  type: REMOVE,
-  book,
-});
+  const books = [];
+  Object.keys(booksFetch).forEach((e) => {
+    books.push({ ...booksFetch[e][0], item_id: e });
+  });
+  dispatch({
+    type: GET_BOOKS,
+    payload: books,
+  });
+};
+
+export const addBook = (book) => async (dispatch) => {
+  await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/z4gzcApO94uwV3qBSQiu/books',
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        item_id: book.item_id,
+        title: book.title,
+        author: book.author,
+        category: book.category,
+      }),
+    }
+  ).then(
+    dispatch({
+      type: ADD,
+      book,
+    })
+  );
+};
+
+export const removeBook = (book) => async (dispatch) => {
+  await fetch(
```

Comment:
- [ ] The same goes here and on all your action creators. Please try to use the method mentioned in the project requirements which is using CreateAsyncThunk and CreateSlice functions from the redux-toolkit library. I have added important links in the above review so, you can use them to implement it in all your action creators 😉